system/jlib/jsmartsock.cpp

```diff
@@ -326,6 +326,9 @@ SocketEndpoint& CSmartSocketFactory::nextEndpoint()
     SmartSocketEndpoint *ss=nextSmartEndpoint();
     if (!ss)
         throw createSmartSocketException(0, "smartsocket failed to get nextEndpoint");
+
+    synchronized block(lock);
+    ss->checkHost(dnsInterval);
```

Comment:
I think should move this inside nextSmartEndpoint, but make it conditional,
e.g. nextSmartEndpoint(bool validate)

call it with true here, and look at other places that call it, which manually call checkHost themselves, so wouldn't need to if nextSmartEndpoint has done already.

Reply:
Moved